iceoryx_hoofs/testing/include/iceoryx_hoofs/testing/logger.hpp

```diff
@@ -33,15 +33,22 @@ class Logger : public platform::TestingLoggerBase
     using Base = platform::TestingLoggerBase;
 
   public:
-    static void init();
+    static void init() noexcept;
 
-    void clearLogBuffer();
+    void clearLogBuffer() noexcept;
 
-    void printLogBuffer();
+    void printLogBuffer() noexcept;
 
-    static uint64_t getNumberOfLogMessages();
+    static uint64_t getNumberOfLogMessages() noexcept;
 
-    static std::vector<std::string> getLogMessages();
+    static std::vector<std::string> getLogMessages() noexcept;
+
+    /// @brief Checks if the the LogLevel is above the minimal supported LogLevel compiled into the binary
+    /// @note This can be used in tests which check for a specific log output
+    static inline constexpr bool doesLoggerSupportLogLevel(const log::LogLevel logLevel) noexcept
```

Comment:
Please move the implementation into the inl file or cpp file

Reply:
you catched me again. I didn't want to move this into the cpp file since it is a `constexpr` and since it is a oneliner I also did not want to create a separate file just for this.

In general I would propose to loose our rules in this case and not be more catholic than the pope. But like I said in a previous comment, I won't if you still like this to be moved to an `inl` file.

Reply:
I am fine with this here but if the static code analysis catches you, you will owe me a [Octomore 12.3](https://www.thewhiskybank.de/shop/octomore-12-2-2304.html)

MMmmmhhh I have an idea for a new static code analysis tool we urgently require. I would call it `catch-a-bob`

Reply:
Deal. You get the Octomore if clang-tidy or Axivion is complaining about this